### ISSUE TITLE
Password policies are not enforced correctly

### ISSUE THREAD

**Issue by infthi** [nextcloud/server#16465]
TL;DR: Minor client-side bugs that prevent users from using empty passwords (convenient for a small local dev setup) even if such passwords are allowed by password policies. Web developer tools do help.

### Steps to reproduce
1. Set up nextcloud and sign in as an admin user
2. Go to settings-administration-security
3. Set up minimal password length to 0 and disable other password policies
4. Attempt to create a new user with password and email fields empty
5. Go to Settings->Personal->Security and attempt to change your password to an empty one.
6. Use code from https://github.com/nextcloud/server/blob/master/settings/js/security_password.js#L28L58 to set an empty password through JS console
7. Log out and attempt to log in with an empty password

### Expected behaviour
Step 4: Should succeed, i.e. a user with an empty password should be created successfully.
Step 5: Should succeed, i.e. active user's password should be set to an empty one
Step 7: Should succeed, i.e. you should be able to log in using the empty password set on the step 5 or step 6.

### Actual behaviour
Step 4: Fails, input id="newuserpassword" is marked as required="required" (settings/src/components/userList.vue#L66). Removal of that attribute leads to some other factors preventing from successful initialization of a user with an empty password.
Step 5: Fails due to a check in security_password.js#L27
Step 7: Fails, input id="password" is marked as required (components/login/LoginForm.vue#L76). Login succeeds if that attribute is removed.

Probably there are other places with excessive client-side checks not following the configured policies.

### Server configuration

**Nextcloud version:** 16.0.3

### Client configuration

**Browser:** Firefox 68.0

**Comment by rullzer** [comment]
Well empty passwords are not supported in general.
If anything we should fix places that accept them.

**Comment by nursoda** [comment]
Yet there is/was a feature to be able to create users without setting a password so that they may set a password for themselves. This is a typical situation for the usecase I have: A consulting company (e.g. lawyer, tax consultant) provides limited access to a NC workspace but neither wants to allow self-registration nor does have central user directory. It even results in a different kind of welcome mail to the user. However, it seems that this worked in NC 16 but does no longer in 17.0.1. I just opened a bug for that: https://github.com/nextcloud/server/issues/18127

**Comment by szaimen** [comment]
Is this Issue still valid? If not, please close this issue. Thanks! :)

**Comment by nursoda** [comment]
Can't tell if this issue is still relevant but for my usecases, user accounts without password are not an option, not even in "intranet/household usage", but that's my personal opinion / security mindset. But as the NC webinterface will often be facing public internet, in my opinion there should be a big one-time and/or permanent warning if the minimum password length is set to 0 (or better: to < 8) characters.

**Comment by szaimen** [comment]
> there should be a big one-time and/or permanent warning if the minimum password length is set to 0 (or better: to < 8) characters.

This is out of scope for this ticket.

---

cc @nextcloud/security do we want to allow empty passwords? If not, please close this issue. Thanks! :)

**Comment by szaimen** [comment]
Related: https://github.com/nextcloud/server/issues/20073
